File: frontend/src/lib/api/results.ts
/*
 1. FILE PURPOSE: API calls to fetch results — flags, script detail, collusion graph.
 2. RESPONSIBILITIES:
    - getFlags(batchId); getScript(id); getCollusionGraph(batchId).
 3. PLANNED CONTENTS: async functions using the shared axios client.
 4. INPUTS / OUTPUTS: Inputs: batch/script ids. Outputs: FlagOut[], ScriptOut, graph JSON.
 5. DEPENDS ON / USED BY: lib/api/client, types; used by useFlags + review/scripts pages.
*/

File: backend/app/api/routes/results.py
# NOTE: Calls services/evaluation_service.py to fetch evaluated sheets.
# 1. FILE PURPOSE: API contract for the frontend to fetch the evaluated answer sheets produced by Phase 2.
# 2. RESPONSIBILITIES:
#    - GET /results — list evaluated scripts in a batch with totals + percentages.
#    - GET /results/{script_id} — full evaluated sheet: question-wise marks, feedback, deduction reasons.
#    - GET /results/{script_id}/answers — per-answer detail (student text, key, similarity, mark).
# 3. PLANNED CONTENTS: FastAPI APIRouter. Returns ScriptEvaluation / AnswerResult from SQLite/JSON store.
# 4. INPUTS / OUTPUTS: Inputs: HTTP GET parameters. Outputs: structured evaluated-sheet JSON.
# 5. DEPENDS ON / USED BY: FastAPI, SQLite, JSON store; consumed by the frontend Results pages.
